doc/developer_guide/psyir.rst

```diff
@@ -168,6 +168,37 @@ Iteration construct
 .. autoclass:: psyclone.psyGen.Loop
     :members:
 
+Ranges
```

Comment:
Could you please update the user guide as appropriate.

Reply:
That's a difficult one as I'm not entirely clear about the distinction between user- and developer-guides when it comes to the PSyIR. I've simply mentioned it as a node type in the UG.

Reply:
Agreed, it is unclear. A mention if fine given the way we currently document.